extension/goext/sync.go

```diff
@@ -0,0 +1,43 @@
+// Copyright (C) 2017 NTT Innovation Institute, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package goext
+
+import "time"
+
+// Event is an event from Sync
+type Event struct {
+	Action   string
+	Key      string
+	Data     map[string]interface{}
+	Revision int64
+}
+
+// Node is a node from Sync
+type Node struct {
+	Key      string
+	Value    string
+	Revision int64
+	Children []*Node
+}
+
+// ISync is an interface to sync in Gohan
+type ISync interface {
+	IEnvironmentSupport
+
+	Fetch(path string) (*Node, error)
+	Delete(path string, prefix bool) error
+	Watch(path string, timeout time.Duration, revision int64) ([]*Event, error)
```

Comment:
we should use a method similar to [WatchContext](https://github.com/cloudwan/gohan/blob/master/sync/sync.go#L44). we could use the context to trigger interrupts (execution timeouts)